blocks_vertical/data.js

```diff
@@ -526,20 +526,68 @@ Blockly.Constants.Data.CUSTOM_CONTEXT_MENU_GET_VARIABLE_MIXIN = {
 
         option.callback =
             Blockly.Constants.Data.VARIABLE_OPTION_CALLBACK_FACTORY(this,
-                option.text);
+                option.text, 'VARIABLE');
         options.push(option);
       }
     } else {
       var renameOption = {
         text: Blockly.Msg.RENAME_VARIABLE,
         enabled: true,
-        callback: Blockly.Constants.Data.RENAME_OPTION_CALLBACK_FACTORY(this)
+        callback: Blockly.Constants.Data.RENAME_OPTION_CALLBACK_FACTORY(this, 'VARIABLE')
       };
       var name = this.getField('VARIABLE').text_;
       var deleteOption = {
         text: Blockly.Msg.DELETE_VARIABLE.replace('%1', name),
         enabled: true,
-        callback: Blockly.Constants.Data.DELETE_OPTION_CALLBACK_FACTORY(this)
+        callback: Blockly.Constants.Data.DELETE_OPTION_CALLBACK_FACTORY(this, 'VARIABLE')
+      };
+      options.push(renameOption);
+      options.push(deleteOption);
+    }
+  }
+};
+
+/**
+ * Mixin to add a context menu for a data_list block.  It adds one item for
+ * each list defined on the workspace.
+ * @mixin
+ * @augments Blockly.Block
+ * @package
+ * @readonly
+ */
+Blockly.Constants.Data.CUSTOM_CONTEXT_MENU_GET_LIST_MIXIN = {
+  /**
+   * Add context menu option to create getter block for the loop's variable.
+   * (customContextMenu support limited to web BlockSvg.)
+   * @param {!Array} options List of menu options to add to.
+   * @this Blockly.Block
+   */
+  customContextMenu: function(options) {
+    if (this.isCollapsed()) {
+      return;
+    }
+    if (!this.isInFlyout) {
+      var variablesList = this.workspace.getVariablesOfType(Blockly.LIST_VARIABLE_TYPE);
+      for (var i = 0; i < variablesList.length; i++) {
+        var option = {enabled: true};
+        option.text = variablesList[i].name;
+
+        option.callback =
+            Blockly.Constants.Data.VARIABLE_OPTION_CALLBACK_FACTORY(this,
+                option.text, 'LIST');
+        options.push(option);
+      }
+    } else {
+      var renameOption = {
+        text: Blockly.Msg.RENAME_VARIABLE,
```

Comment:
This should be changed to `Blockly.Msg.RENAME_LIST`, and the appropriate message should be added in `/msg/messages.js`. Right now it displays as "rename variable" instead of "rename list"!

Reply:
See I wasn't sure about this one - that was my intuition at first too but checked the dropdown for the other list blocks and noticed they had the same naming convention so I kept it unchanged for consistency. What do you think?

<img width="287" alt="screen shot 2018-09-13 at 17 46 28" src="https://user-images.githubusercontent.com/16880268/45496030-356e0d80-b77d-11e8-899f-b64e6432b132.png">

Reply:
Oh, interesting. I think that that's a bug and should be changed too. I don't think most users think of lists as another type of variable; that is, both variables and lists are things which store data (which is why they're in the Data category), but they're distinct.

Reply:
Oh also, I'm 99.5% sure (although I can't check at the moment) that Scratch 2.0 has a "delete list" option - not "delete variable".

Reply:
You are correct. In Scratch 2.0 you can find "rename variable" and "delete variable" for variables whereas for lists the menu only contains "delete list". I can make the changes and also look into changing the text in the dropdown.